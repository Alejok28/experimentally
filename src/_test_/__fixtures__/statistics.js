export const statistics = {
  "viewCount": "28640082",
  "likeCount": "556518",
  "dislikeCount": "6249",
  "favoriteCount": "0",
  "commentCount": "12663"
}