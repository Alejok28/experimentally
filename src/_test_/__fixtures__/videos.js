export const videos = [
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/n8La5FHPnJZ9SjefY7NHLK55KlY\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "z2VZ_OGz0RU"
    },
    "snippet": {
      "publishedAt": "2017-03-22T10:01:08.000Z",
      "channelId": "UCftwRNsjfRo08xYE31tkiyw",
      "title": "Ryan Reynolds &amp; Jake Gyllenhaal Answer the Web&#39;s Most Searched Questions | WIRED",
      "description": "'Life' stars Ryan Reynolds and Jake Gyllenhaal waste some time with WIRED and answer the Internet's most searched questions about themselves. Does Ryan ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/z2VZ_OGz0RU/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "WIRED",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/KPeWEEZzKXm-pfv-1ywpZ2WOtPE\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "F9CrRG6j2SM"
    },
    "snippet": {
      "publishedAt": "2019-08-05T09:00:02.000Z",
      "channelId": "UCfkXDY7vwkcJ8ddFGz8KusA",
      "title": "[M/V] SEVENTEEN(세븐틴) - HIT",
      "description": "13명의 뚜렷한 음악 색을 바탕으로 세븐틴 표 노래와 무대를 전 세계에 알리며 독보적인 독창성을 보여주고 있는 자체 제작 아이돌 세븐틴이 또...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/F9CrRG6j2SM/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "SEVENTEEN",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/id3uz64VI_Frb6lxlzAYDG8yCuw\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "A2FsgKoGD04"
    },
    "snippet": {
      "publishedAt": "2017-05-25T21:58:58.000Z",
      "channelId": "UCRijo3ddMTht_IHyNSNXpNQ",
      "title": "Real Life Trick Shots | Dude Perfect",
      "description": "Trick shots should be an everyday thing! Thanks to Kingsford for sponsoring this video! ▻ Click HERE to subscribe to Dude Perfect! http://bit.ly/SubDudePerfect ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/A2FsgKoGD04/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Dude Perfect",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/mGGEbULKJWR5g9X1nJ2MGKv0MKo\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "t_qn-f7XfJo"
    },
    "snippet": {
      "publishedAt": "2019-04-26T20:58:47.000Z",
      "channelId": "UCte53PewI8_jYpBYn071MqQ",
      "title": "Sech - Otro Trago ft. Darell (Video Oficial)",
      "description": "Escúchalo en Spotify: https://open.spotify.com/track/1Ej96GIBCTvgH7tNX1r3qr?si=xhlKQ1ZGR6iT5xnErFxojA Producido por Cinema Entertainment Sigue a ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/t_qn-f7XfJo/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Sech",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/1szGxIC2JpHOZayLDDyWY1wRMRg\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "VRJmcxCrAOA"
    },
    "snippet": {
      "publishedAt": "2018-01-22T22:55:24.000Z",
      "channelId": "UCRijo3ddMTht_IHyNSNXpNQ",
      "title": "Real Life Trick Shots 2 | Dude Perfect",
      "description": "Trick shots should be an everyday thing! Thanks to Kay Jewelers for sponsoring this video! Win Valentine's Day by going to http://dudeperfect.com/KayJewelers ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/VRJmcxCrAOA/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Dude Perfect",
      "liveBroadcastContent": "none"
    }
  },
  {
    "kind": "youtube#searchResult",
    "etag": "\"nlUZBA6NbTS7q9G8D1GljyfTIWI/0WX426ECapOMYBfaUd1p8CnScQ8\"",
    "id": {
      "kind": "youtube#video",
      "videoId": "rtgY1q0J_TQ"
    },
    "snippet": {
      "publishedAt": "2014-09-22T23:08:57.000Z",
      "channelId": "UCUsN5ZwHx2kILm84-jPDeXw",
      "title": "Key &amp; Peele - Gay Wedding Advice",
      "description": "In preparation for their cousin's marriage to another man, a family asks a gay man questions about what the wedding ceremony will be like and for \"advice\" about ...",
      "thumbnails": {
        "default": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/default.jpg",
          "width": 120,
          "height": 90
        },
        "medium": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/mqdefault.jpg",
          "width": 320,
          "height": 180
        },
        "high": {
          "url": "https://i.ytimg.com/vi/rtgY1q0J_TQ/hqdefault.jpg",
          "width": 480,
          "height": 360
        }
      },
      "channelTitle": "Comedy Central",
      "liveBroadcastContent": "none"
    }
  }
]