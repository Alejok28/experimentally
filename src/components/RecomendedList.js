import React from 'react';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import CircularProgress from '@material-ui/core/CircularProgress';
import Tooltip from '@material-ui/core/Tooltip';
import { escapeHtml } from '../utils/functions';

export default function RecomendedList({ videos = [], loading, handleClick }) {
  return (
    <Container style={{ margin:'40px 0'}}>
      {loading ? (
        <CircularProgress color="secondary" />
      ) : (
        <div style={{ display: 'flex', justifyContent: 'space-between', overflow: 'scroll'}}>
          {videos.map(video => {
            const { snippet, id } = video;
            return (
              <Grid key={id.videoId} item xs={12} style={{ margin: '10px'}}>
                <Tooltip title={escapeHtml(snippet.title)}>
                  <img
                    style={{ cursor: 'pointer'}}
                    src={snippet.thumbnails.default.url}
                    alt={id.videoId}
                    onClick={() => handleClick(video)}
                  />
                </Tooltip>
              </Grid>
            )
          })}
        </div>
      )
    }
    </Container>
  )
}