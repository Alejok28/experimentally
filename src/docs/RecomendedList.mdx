---
name: RecommendedList
route: /components/RecomendedList
menu: Components
---

import { Playground } from 'docz';
import RecomendedList from '../components/RecomendedList';
import { videos } from '../_test_/__fixtures__/videos';


# RecomendedList
RecommendedList is a component that shows a list of the first six videos that match with the search, otherwise if there is no search the component will show the recommended videos.

## Properties

| Name          | Type     | Required     |Description                                |
|:--------------|:--------:|:------------:|:------------------------------------------|
| videos        | Array    | false = []   | Array with the info commig from YoutubeApi|
| loading       | Bool     | false = false | Prop to indicate if the api has already  responded|
| handleClick   | Func     | true       | This function is in charge for selecting a video to play in the VideoPlayer component |


## Basic usage

### Waiting for the request response

<Playground>
  <RecomendedList
    loading={true}
    handleClick={video => console.log(video)}
  />
</Playground>

### When the request is successful
<Playground>
  <RecomendedList
    videos={videos}
    loading={false}
    handleClick={video => console.log(video)}
  />
</Playground>


