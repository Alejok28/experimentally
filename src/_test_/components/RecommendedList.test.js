import React from 'react';
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import RecomendedList from '../../components/RecomendedList';
import { videos } from '../__fixtures__/videos';

afterEach(cleanup);

test('renders loading when ', () => {
  const { container } = render(<RecomendedList loading={true} />);
  const icon = container.querySelector('svg');
  expect(icon).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

test('renders when videos props is passed', () => {
  const { container } = render(<RecomendedList videos={videos} />);
  const images = container.querySelectorAll('img');
  expect(images).toHaveLength(6);
  expect(container).toMatchSnapshot();
});
