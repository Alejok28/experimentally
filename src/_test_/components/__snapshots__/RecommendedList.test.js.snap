// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading when  1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`renders when videos props is passed 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-root-24 MuiContainer-maxWidthLg"
  >
    <div
      class="makeStyles-listContainer-26"
    >
      <div
        class="MuiGrid-root makeStyles-grid-35 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="Ryan Reynolds &amp; Jake Gyllenhaal Answer the Web&#39;s Most Searched Questions | WIRED"
          class="makeStyles-img-36"
          src="https://i.ytimg.com/vi/z2VZ_OGz0RU/default.jpg"
          title="Ryan Reynolds & Jake Gyllenhaal Answer the Web's Most Searched Questions | WIRED"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-grid-35 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="[M/V] SEVENTEEN(세븐틴) - HIT"
          class="makeStyles-img-36"
          src="https://i.ytimg.com/vi/F9CrRG6j2SM/default.jpg"
          title="[M/V] SEVENTEEN(세븐틴) - HIT"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-grid-35 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="Real Life Trick Shots | Dude Perfect"
          class="makeStyles-img-36"
          src="https://i.ytimg.com/vi/A2FsgKoGD04/default.jpg"
          title="Real Life Trick Shots | Dude Perfect"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-grid-35 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="Sech - Otro Trago ft. Darell (Video Oficial)"
          class="makeStyles-img-36"
          src="https://i.ytimg.com/vi/t_qn-f7XfJo/default.jpg"
          title="Sech - Otro Trago ft. Darell (Video Oficial)"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-grid-35 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="Real Life Trick Shots 2 | Dude Perfect"
          class="makeStyles-img-36"
          src="https://i.ytimg.com/vi/VRJmcxCrAOA/default.jpg"
          title="Real Life Trick Shots 2 | Dude Perfect"
        />
      </div>
      <div
        class="MuiGrid-root makeStyles-grid-35 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <img
          alt="Key &amp; Peele - Gay Wedding Advice"
          class="makeStyles-img-36"
          src="https://i.ytimg.com/vi/rtgY1q0J_TQ/default.jpg"
          title="Key & Peele - Gay Wedding Advice"
        />
      </div>
    </div>
  </div>
</div>
`;
